src/pages/signin/PasswordForm.js

```diff
@@ -52,6 +57,20 @@ class PasswordForm extends React.Component {
         };
     }
 
+    componentDidMount() {
+        if (!this.input) {
+            return;
+        }
+        this.input.focus();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.visible || !this.props.visible) {
+            return;
+        }
+        this.input.focus();
+    }
```

Comment:
NAB, this logic looks very similar to the logic in `LoginForm`. I'm not sure how common this pattern will be after this PR, but maybe there is some way to abstract it (instead of copy paste everywhere this trick is needed).

Reply:
Yeah, good Idea

Reply:
Thoughts on how we might be able to do it? Or if it's worth doing?

Reply:
One way or another we have to manually set the focus when page becomes visible. 
So what I can think of is that we pass a callback from the `ToggleVisibilityView` which we call here to set the ref of the input.
and Move the whole logic to focus the input in the HOC. 

we set the focus whenever page becomes visible and the ref is set.

so in this component we just need to call something like.
```this.props.setRefForToggleVisibility(this.textinput);```

Reply:
Could forward the ref and do 

```jsx
<WrappedComponent
    ref={el => this.focusableElement = el}
/>
```

Reply:
I don't think it would be as straight as this. we will have to forwardRef the from LoginForm. so in future any component that is using this HOC has to manually forward the ref. 
Where calling this.props.setRefForToggleVisibility(this.textinput); would be much better.

`WrappedComponent` is not a Textinput. It could be group of elments and like a page. So forwarding a ref does not sound good. It is unclear ,whose ref will it be on the page etc.

Reply:
Hmm well there is only one "ref" that we are concerned about in the HOC right?

>so in future any component that is using this HOC has to manually forward the ref

Seems pretty much the same if you have to do `setRefForToggleVisibility(this.textinput)` - otherwise there will be no ref.

But maybe it is not very straightforward to do either - and in this case might just be better to be explicit. Want to pivot and just add some comments about why the logic is required? We can always revisit this later if the pattern pops up a lot.

Reply:
Overall, I think it is not that much helpful atm. We do need to use isVisible prop on both LoginForm and passwordForm.

Reply:
K let's just leave this for now.